pkg/pgmodel/querier/querier.go

```diff
@@ -30,13 +31,20 @@ type Reader interface {
 	Read(*prompb.ReadRequest) (*prompb.ReadResponse, error)
 }
 
+type QueryHints struct {
```

Comment:
note we don't control *storage.SelectHints so we add our own hints that are necessary for our pushdowns